src/hotspot/share/utilities/debug.cpp

```diff
@@ -302,7 +306,22 @@ void report_fatal(const char* file, int line, const char* detail_fmt, ...)
 
   print_error_for_unit_test("fatal error", detail_fmt, detail_args);
 
-  VMError::report_and_die(Thread::current_or_null(), context, file, line, "fatal error", detail_fmt, detail_args);
+  VMError::report_and_die(error_type, "fatal error", detail_fmt, detail_args,
+                          Thread::current_or_null(), NULL, NULL, context,
+                          file, line, 0);
+}
+
+void report_fatal(const char* file, int line, const char* detail_fmt, ...) {
+  va_list detail_args;
+  va_start(detail_args, detail_fmt);
+  report_fatal_impl(INTERNAL_ERROR, file, line, detail_fmt, detail_args);
+  va_end(detail_args);
+}
```

Comment:
Do we really need this API? Can't we just update:

```
#define fatal(...)                                                                \
do {                                                                              \
  TOUCH_ASSERT_POISON;                                                            \
  report_fatal(INTERNAL_ERROR, __FILE__, __LINE__, __VA_ARGS__);                                  \
  BREAKPOINT;                                                                     \
} while (0)

```

Reply:
As the fatal() macro seems to be the only user of the existing `report_fatal` then yes I think we can simplify this as you suggest. I had thought it was used directly from elsewhere.

Thanks for the suggestion.